src/js/components/FrameworkConfigurationForm.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component, PropTypes } from "react";
+import classNames from "classnames";
+import DefaultTitleField
+  from "react-jsonschema-form/lib/components/fields/TitleField";
+import SchemaForm from "react-jsonschema-form";
+import TabButton from "#SRC/js/components/TabButton";
+import TabButtonList from "#SRC/js/components/TabButtonList";
+import Tabs from "#SRC/js/components/Tabs";
+import Util from "#SRC/js/utils/Util";
+import JSONEditor from "#SRC/js/components/JSONEditor";
+import PageHeaderNavigationDropdown
+  from "#SRC/js/components/PageHeaderNavigationDropdown";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import SchemaField from "#SRC/js/components/SchemaField";
+
+export default class FrameworkConfigurationForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errorSchema: null
+    };
+  }
+
+  getFormattedSectionLabel(label) {
+    return label
+      .toLowerCase()
+      .split("_")
+      .map(word => word[0].toUpperCase() + word.slice(1))
+      .join(" ");
+  }
+
+  getFirstErrorFieldPath(tabName, errorSchema) {
+    const { packageDetails } = this.props;
+    const schema = packageDetails.config;
+
+    const fieldsWithErrors = Object.keys(errorSchema[tabName]).filter(field => {
+      if (
+        errorSchema[tabName][field].__errors &&
+        errorSchema[tabName][field].__errors.length > 0
+      ) {
+        return true;
+      }
+
+      return false;
+    });
+
+    const fieldToFocus = Object.keys(
+      schema.properties[tabName].properties
+    ).find(field => {
+      return fieldsWithErrors.includes(field);
+    });
+
+    return [tabName, fieldToFocus];
+  }
+
+  handleBadgeClick(tabName, event) {
+    const { errorSchema } = this.state;
+
+    const newFocusFieldPath = this.getFirstErrorFieldPath(tabName, errorSchema);
+
+    this.props.onFocusFieldPathChange(newFocusFieldPath);
+
+    event.stopPropagation();
```

Comment:
I'm going to guess that if you put `event.stopPropagation()` at the top doesn't work as expected?

Reply:
I've moved this to the top, it works either way, but nicer at the top so I'll put it there.